CMakeLists.txt

```diff
@@ -75,26 +75,35 @@ include(CheckCXXCompilerFlag)
 include(CheckIncludeFileCXX)
 include(GNUInstallDirs)
 
-# Check SYCL support by the compiler
-check_cxx_compiler_flag("-fsycl" _fsycl_option)
-if (_fsycl_option)
-    CHECK_INCLUDE_FILE_CXX("sycl/sycl.hpp" _sycl_header "-fsycl")
-    if (NOT _sycl_header)
-        CHECK_INCLUDE_FILE_CXX("CL/sycl.hpp" _sycl_header_old "-fsycl")
+# Detect SYCL support
+if (NOT DEFINED ONEDPL_BACKEND OR ONEDPL_BACKEND MATCHES "^(dpcpp|dpcpp_only)$")
+    message(STATUS "Checking for SYCL support")
+
+    set(FSYCL_OPTION "-fsycl")
+    check_cxx_compiler_flag(${FSYCL_OPTION} _fsycl_option)
+    if (_fsycl_option)
+        set(FSYCL_OPTION_IF_SUPPORTED ${FSYCL_OPTION})
     endif()
-    if (_sycl_header OR _sycl_header_old)
-        set(_sycl_support TRUE)
+
+    CHECK_INCLUDE_FILE_CXX("sycl/sycl.hpp" SYCL_HEADER ${FSYCL_OPTION_IF_SUPPORTED})
+    if (NOT SYCL_HEADER)
+        CHECK_INCLUDE_FILE_CXX("CL/sycl.hpp" SYCL_HEADER_OLD ${FSYCL_OPTION_IF_SUPPORTED})
     endif()
-endif()
 
-# Exit with an error when DPCPP backend is explicitly set and SYCL is not supported
-if (ONEDPL_BACKEND MATCHES "^(dpcpp|dpcpp_only)$" AND NOT _sycl_support)
-    message(FATAL_ERROR "${CMAKE_CXX_COMPILER} doesn't support -fsycl option or sycl.hpp is not available. It is required if ONEDPL_BACKEND=${ONEDPL_BACKEND}")
+    if (SYCL_HEADER OR SYCL_HEADER_OLD)
+        message(STATUS "SYCL is supported")
+    else()
+        if (ONEDPL_BACKEND MATCHES "^(dpcpp|dpcpp_only)$")
+            message(FATAL_ERROR "SYCL is not supported. It is required for ONEDPL_BACKEND=${ONEDPL_BACKEND}")
+        else()
+            message(STATUS "SYCL is not supported")
+        endif()
+    endif()
 endif()
 
 # Set the default backend if one has not been explicitly provided
-if (NOT ONEDPL_BACKEND)
-    if (_sycl_support)
+if (NOT DEFINED ONEDPL_BACKEND)
+    if (SYCL_SUPPORT)
```

Comment:
I'm confused where this variable `SYCL_SUPPORT` is coming from. Should this condition instead be checking if `FSYCL_OPTION_IF_SUPPORTED` is set to a non-empty value?

Reply:
Good catch. It was rather a refactoring mistake, which could be avoided by proper testing. Now `SYCL_SUPPORT` is set.